nbgitpuller/handlers.py

```diff
@@ -152,12 +154,14 @@ def get(self):
         subPath = self.get_argument('subpath', None) or \
                   self.get_argument('subPath', '.')
         app = self.get_argument('app', app_env)
+        parent_reldir = os.getenv('NBGITPULLER_PARENTPATH', '')
+        targetpath = self.get_argument('targetpath', None) or \
+                     self.get_argument('targetPath', repo.split('/')[-1])
 
         if urlPath:
             path = urlPath
         else:
-            repo_dir = repo.split('/')[-1]
-            path = os.path.join(repo_dir, subPath)
+            path = os.path.join(parent_reldir, targetpath, subPath)
```

Comment:
What if `targetPath` doesn't exist? Will this throw an error, or will it create the path? If it throws an error, will it be a helpful error that a user could understand and surface to an admin?

Reply:
Just from using it, it will create the `targetPath`. I think this is a consequence of the behavior of specifying a target directory when using git. From a quick look, there doesn't seem to be any code that explicitly creates the directory.

Reply:
perfect, that sounds reasonable to me then 👍

Reply:
@albertmichaelj thanks for checking- that makes sense, `git clone` creates whatever parent directories it needs.